src/main/java/core/basesyntax/ReadWriteList.java

```diff
@@ -10,16 +10,24 @@ public class ReadWriteList<E> {
     private ReadWriteLock lock = new ReentrantReadWriteLock();
 
     public void add(E element) {
-        // write your code here
+        try {
+            lock.writeLock().lock();
+            list.add(element);
+        } finally {
+            lock.writeLock().unlock();
+        }
     }
 
     public E get(int index) {
-        // write your code here
-        return null;
+        try {
+            lock.readLock().lock();
+            return list.get(index);
+        } finally {
+            lock.readLock().unlock();
+        }
     }
 
     public int size() {
-        // write your code here
-        return 0;
+        return list.size();
```

Comment:
Access to the `size` method of the underlying list should also be protected with a read lock to ensure thread safety. Without this, there is a possibility of getting an inconsistent size if another thread is modifying the list.